src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -371,6 +374,96 @@ def image_mask_iterator():
     return nwbfile
 
 
+def add_fluorescence(segmentation_extractor: SegmentationExtractor, nwbfile: NWBFile, metadata: dict) -> NWBFile:
+    """
+    Adds the fluorescence specificied by the metadata to the nwb file.
+    The fluorescence that is added is the one located in metadata["Ophys"]["Fluorescence"]
+
+    Parameters
+    ----------
+    nwbfile : NWBFile
+        An previously defined -in memory- NWBFile.
+    metadata : dict
+        The metadata in the nwb conversion tools format.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the fluorescence added.
+    """
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    fluorescence_metadata = metadata_copy["Ophys"]["Fluorescence"]
+    fluorescence_name = fluorescence_metadata["name"]
+
+    ophys = get_module(nwbfile, "ophys")
+    if fluorescence_name in ophys.data_interfaces:
```

Comment:
What if the fluorescence module already exists but there are still some traces to add?

Reply:
ah I see your point, the early return makes no sense here. I should also add a check to not add those traces again if they already exist in fluorescence.